src/main/java/com/list/todo/services/UserService.java

```diff
@@ -57,7 +57,14 @@ public UserService(UserRepository repository, TodoListRepository todoListReposit
     }
 
     public UserSummary getUserInfo(UserPrincipal user) {
-        return new UserSummary(user.getUsername(), user.getName(), user.getEmail());
+        Optional<User> currentUser = userRepository.findById(user.getId());
+        UserSummary userSummary = null;
+
+        if (currentUser.isPresent()) {
+            // TODO: url задавать в таком виде, либо брать из файла пропертей? Если брать из файла, то как поступать с сущностями, в которые нельзя добавить поле для пропертей, иначе оно пойдет в базу
+            userSummary = new UserSummary(currentUser.get().getUsername(), currentUser.get().getName(), currentUser.get().getEmail(), "https://www.gravatar.com/avatar/" + currentUser.get().getGravatarHash());
+        }
+        return userSummary;
```

Comment:
please consider:
```java
return userRepository.findById(user.getId())
.map(currentUser -> 
new UserSummary(currentUser.get().getUsername(), currentUser.get().getName(), currentUser.get().getEmail(), "https://www.gravatar.com/avatar/" + currentUser.get().getGravatarHash()))
.orElse(null);

```
instead of
```java
        Optional<User> currentUser = userRepository.findById(user.getId());
        UserSummary userSummary = null;

        if (currentUser.isPresent()) {
            // TODO: url задавать в таком виде, либо брать из файла пропертей? Если брать из файла, то как поступать с сущностями, в которые нельзя добавить поле для пропертей, иначе оно пойдет в базу
            userSummary = new UserSummary(currentUser.get().getUsername(), currentUser.get().getName(), currentUser.get().getEmail(), "https://www.gravatar.com/avatar/" + currentUser.get().getGravatarHash());
        }
        return userSummary;
```